### PR TITLE
Refactor otlp exporter to upcoming auth changes

### PR DESCRIPTION
**Pull request by pavankrish123**
Description:
Updated oltpexporter for upcoming client auth extension changes

Link to tracking Issue:
#3287

Testing:
Unit tests

Documentation:
Unit tests

### CODE REVIEW

File: exporter/otlpexporter/otlp.go [hunk 1]
```diff
@@ -93,7 +100,7 @@ type grpcSender struct {
 	callOptions    []grpc.CallOption
 }
 
-func newGrpcSender(config *Config) (*grpcSender, error) {
+func newGrpcSender(config *Config, _ map[config.ComponentID]component.Extension) (*grpcSender, error) {
```
Comment (bogdandrutu) [inline exporter/otlpexporter/otlp.go:103]: not necessary now.

File: exporter/otlpexporter/factory_test.go [hunk 2]
```diff
@@ -57,11 +58,11 @@ func TestCreateMetricsExporter(t *testing.T) {
...
 func TestCreateTracesExporter(t *testing.T) {
 	endpoint := testutil.GetAvailableLocalAddress(t)
-
 	tests := []struct {
-		name     string
-		config   Config
-		mustFail bool
+		name            string
+		config          Config
+		mustFail        bool
```
Comment (bogdandrutu) [inline exporter/otlpexporter/factory_test.go:64]: s/mustFail/mustFailOnCreate

File: exporter/otlpexporter/config_test.go [hunk 3]
```diff
@@ -82,11 +83,7 @@ func TestLoadConfig(t *testing.T) {
 					Timeout:             30 * time.Second,
 				},
 				WriteBufferSize: 512 * 1024,
-				PerRPCAuth: &configgrpc.PerRPCAuthConfig{
-					AuthType:    "bearer",
-					BearerToken: "some-token",
-				},
-				BalancerName: "round_robin",
```
Comment (bogdandrutu) [inline exporter/otlpexporter/config_test.go:89]: Why this change? It is still using the `configgrpc.PerRPCAuthConfig`

**Review by bogdandrutu — COMMENTED** [review]
LGTM except some comments.

File: exporter/otlpexporter/config_test.go (hunk 3 above)
Comment (pavankrish123) [reply exporter/otlpexporter/config_test.go:89]: Ah my bad

File: exporter/otlpexporter/otlp.go (hunk 1 above)
Comment (pavankrish123) [reply exporter/otlpexporter/otlp.go:103]: Can we keep it for the time being. The transition will become easy once we modify the ToDIalOption APIs - We don't have to touch a lot test files in the next PR. 

Your call though

File: exporter/otlpexporter/factory_test.go (hunk 2 above)
Comment (pavankrish123) [reply exporter/otlpexporter/factory_test.go:64]: done!

File: exporter/otlpexporter/factory_test.go [hunk 4]
```diff
@@ -178,13 +179,17 @@ func TestCreateTracesExporter(t *testing.T) {
 			factory := NewFactory()
 			creationParams := component.ExporterCreateParams{Logger: zap.NewNop()}
 			consumer, err := factory.CreateTracesExporter(context.Background(), creationParams, &tt.config)
-
-			if tt.mustFail {
+			if tt.mustFailOnCreate {
 				assert.NotNil(t, err)
 			} else {
```
Comment (bogdandrutu) [inline exporter/otlpexporter/factory_test.go:184]: Not your fault, return on the positive branch so remove unnecessary indentation and else branch.

**Review by bogdandrutu — APPROVED** [review]

File: exporter/otlpexporter/factory_test.go (hunk 4 above)
Comment (pavankrish123) [reply exporter/otlpexporter/factory_test.go]: done!

File: exporter/otlpexporter/otlp.go [hunk 5]
```diff
@@ -50,12 +51,18 @@ func newExporter(cfg config.Exporter) (*exporter, error) {
...
+// is the only place we get hold of Extensions which are required to construct auth round tripper.
+func (e *exporter) start(_ context.Context, _ component.Host) error {
+	w, err := newGrpcSender(e.config)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	e.w = w
-	return e, nil
+	return nil
```
Comment (bogdandrutu) [inline exporter/otlpexporter/otlp.go:65]: func (e *exporter) start(_ context.Context, _ component.Host) (err error) {
  e.w, err = newGrpcSender(e.config)
  return err
}

File: exporter/otlpexporter/otlp.go [hunk 6]
```diff
@@ -50,12 +51,17 @@ func newExporter(cfg config.Exporter) (*exporter, error) {
...
+}
+
+// start actually creates the gRPC connection. The client construction is deferred till this point as this
+// is the only place we get hold of Extensions which are required to construct auth round tripper.
+func (e *exporter) start(_ context.Context, _ component.Host) (err error) {
+	e.w, err = newGrpcSender(e.config)
 	if err != nil {
-		return nil, err
+		return err
 	}
```
Comment (bogdandrutu) [inline exporter/otlpexporter/otlp.go:63]: No need to have the if here, just return err :)

Comment (pavankrish123) [reply exporter/otlpexporter/otlp.go:63]: ah good stuff - updating the zipkin one as well.

**Comment by bogdandrutu** [comment]
@pavankrish123 lint error:

```
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector/opentelemetry-collector'
exporter/otlpexporter/otlp.go:51: File is not `gofmt`-ed with `-s` (gofmt)
	
make[2]: *** [Makefile.Common:26: lint] Error 1
```

**Comment by pavankrish123** [comment]
@bogdandrutu sorry missed pushing a commit - should be fixed now.

**Review by bogdandrutu — APPROVED** [review]
